lib/pry/color.rb

```diff
@@ -0,0 +1,97 @@
+module Pry::Color
+  extend self
+  COLORS = {
+    "black"   => 0,
+    "red"     => 1,
+    "green"   => 2,
+    "yellow"  => 3,
+    "blue"    => 4,
+    "purple"  => 5,
+    "magenta" => 5,
+    "cyan"    => 6,
+    "white"   => 7
+  }
+  BRUSH_METHODS = [*COLORS.keys, *["bold", "default", "bright_default", "strip_color"]]
+  UnknownBrushError = Class.new(RuntimeError)
+
+  COLORS.each_pair do |color, value|
+    define_method color do |text, pry=((defined?(_pry_) and _pry_) or Pry)|
```

Comment:
can we extract out this code into its own method, is that possible? since it’s used in multiple places:

```ruby
(defined?(_pry_) and _pry_) or Pry
```

also would it make sense to always have a `_pry_` available but set to `nil` perhaps so that we don’t have to do the `defined?` check? cos: `_pry_ || Pry` looks way cleaner

Reply:
it's hard to make the defined? expression a method because it's called in the calling scope and when used as a singleton, eg Pry::Color.bold "foo", then there is no `_pry_` in scope. we could rescue a NoMethodError on `_pry_` or something like that.